src/Microsoft.ML.Tokenizers/Model/Tiktoken.cs

```diff
@@ -233,7 +311,7 @@ private static (Dictionary<StringSpanOrdinalKey, int>?, Dictionary<int, string>?
         /// <param name="text">The text to encode.</param>
         /// <param name="isSpecialToken">Indicate if the token is a special token.</param>
```

Comment:
token is the string word. can be any word like `dog` or it can be special token like `<|endoftext|>`

Reply:
Is that the parameter `text`, and does that `text` need to represent a single token?  Or does it refer to all tokens within `text`?

Reply:
the text can represent multiple tokens or represent one special token.

Reply:
Ok, so how does `isSpecialToken` apply to 'text` in the case it is multiple tokens?

Reply:
It is a flag telling if the input token is representing a special token or not so the encoder can treat it differently.

Reply:
Ok, so we should probably refer to `text` in the docs.  

Here's my attempt:
```
Indicates if the <paramRef name="text"/> in it's entirety is a special token.  This method will throw if <paramRef name="isSpecialToken"/> is `true` and the specified <paramRef name="text"/> is not a special token.
```

Similarly it looks like the `Count` and `EncodeToIds` just return default values ignoring the text if it's not a special token so they could get a slightly different version of this.
https://github.com/dotnet/machinelearning/blob/164fde09d3739004fc2b027bd1940e1b0afa9ec3/src/Microsoft.ML.Tokenizers/Model/Tiktoken.cs#L313
https://github.com/dotnet/machinelearning/blob/164fde09d3739004fc2b027bd1940e1b0afa9ec3/src/Microsoft.ML.Tokenizers/Model/Tiktoken.cs#L355

I raised this issue because this parameter confused me when adopting the tokenizer.

Reply:
I'm curious as to the use case for setting isSpecialToken to true...?

Reply:
Seems like the only scenario is when someone wants to pass in a single special token string and get the value for that.  If we're making them specify a parameter to this API to do that they might as well just call a different API to do it and avoid the confusing parameter on this API.

I wonder what happens if someone specifies a special token string but forgets to set isSpecialToken?

Reply:
@tarekgh and I talked about this offline - we'll update the docs for these methods in a separate PR and discuss this during API review.